File: day1/src/part1.rs
use std::fs::File;
use std::io::{self, BufRead, BufReader};

fn main() -> io::Result<()> {


    let file = File::open("/home/ersan/aoc2023/day1/input.txt")?;
    let reader = BufReader::new(file);

    let mut lines: Vec<Vec<i32>> = Vec::new();

    for line_result in reader.lines() {
        let mut line = line_result?;
        
        let mut line_numbers: Vec<i32> = Vec::new();

        for character in line.chars() {
            if character.is_numeric(){
                line_numbers.push(character as i32 - 48);
            }
        }

        lines.push(line_numbers);
    
    
    }

    let mut sum = 0;
    for line in lines {
        sum += line[0] * 10 + line[line.len() - 1];
    }
    println!("{}", sum);

    Ok(())
}

File: day8/part1.py
with open("/home/ersan/AOC2023/day8/input.txt", "r") as file:
    lines = file.readlines()

    elements = {line[0:3]: (line[7:10], line[12:15]) for line in lines[2:]}

    directions = lines[0]

    current_pos = "AAA"
    
    step_count = 0
    while current_pos != "ZZZ":
        for direction in directions[:-1]:
            print(current_pos, direction, step_count)
            if current_pos == "ZZZ":
                break
            if direction == "L":
                current_pos = elements[current_pos][0]
            else:
                current_pos = elements[current_pos][1]
            step_count += 1
    
    print(step_count)